internal/util/kubestructs.go

```diff
@@ -0,0 +1,39 @@
+package util
```

Comment:
I'm not sure how I feel about introducing test utility functions inside the main codebase but since these are used by both unit and e2e tests I guess it makes sense. Can we have the file name include `test` in it so that its clear these are for test purposes? Maybe `testcrds` makes sense?

Reply:
What if we add a "util" package into the `test` directory along side `e2e`? That way these testing utility functions can have their own home away from the internal codebase. I tend to agree with you on this that it feels weird.